test/unit/server_test.js

```diff
@@ -24,6 +24,12 @@ describe("server.js tests", function () {
       KinSdk.Config.setAllowHttp(true);
       expect(() => new KinSdk.Server('http://horizon-live.stellar.org:1337')).to.not.throw();
     });
+
+    it("allow insecure server with headers", function () {
```

Comment:
I would add test to verify that headers are added correctly, not sure how exactly here. nock is probably a good candidate.

Reply:
I just did it, why is the current test does not verify that?